lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -92,6 +92,82 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
                                                        operand, vApproximate);
         return success();
       });
+  patterns.onOp(
+      "GridSample", 20,
+      [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        Value input;
+        Value grid;
+        std::string mode;
+        Value interpolationMode;
+        std::string padding;
+        Value paddingMode;
+        int64_t align;
+        Value alignCorners;
+
+        if (binder.tensorOperandAtIndex(input, 0) ||
+            binder.tensorOperandAtIndex(grid, 1) ||
+            binder.tensorResultType(resultType))
+          return failure();
+
+        auto inputTensorType = input.getType().cast<Torch::ValueTensorType>();
+        ArrayRef<int64_t> inputShape = inputTensorType.getSizes();
+        unsigned inputRank = inputShape.size();
+        auto gridTensorType = grid.getType().cast<Torch::ValueTensorType>();
+        ArrayRef<int64_t> gridShape = gridTensorType.getSizes();
+        unsigned gridRank = gridShape.size();
+
+        if (inputRank != 4) {
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "only input rank 4 supported");
+        }
+        if (gridRank != 4) {
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "only grid rank 4 supported");
+        }
+        if (inputShape[0] != gridShape[0]) {
+          return rewriter.notifyMatchFailure(
+              binder.op, "N must be same for input and grid");
+        }
+        if (gridShape[3] != 2) {
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "gridShape[3] ,must be 2");
+        }
+
+        if (binder.customOpNameStringAttr(mode, "mode", "bilinear"))
+          return rewriter.notifyMatchFailure(binder.op, "mode bind failure");
+        if (mode != "bilinear")
+          return rewriter.notifyMatchFailure(
+              binder.op, "currently only mode : bilinear supported");
+        interpolationMode = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getType<Torch::IntType>(),
+            rewriter.getIntegerAttr(rewriter.getIntegerType(64), 0));
+
+        if (binder.customOpNameStringAttr(padding, "padding_mode", "zeros"))
```

Comment:
You should do all of this validation before invoking a `rewriter.create` function.

Reply:
yes that makes more sense